src/lightcurvelynx/math_nodes/scipy_random.py

```diff
@@ -212,3 +213,104 @@ def __init__(self, dist, **kwargs):
         else:
             raise ValueError("No logpdf function detected.")
         super().__init__(self.dist_obj, **kwargs)
+
+
+class ScipyRandomDist(FunctionNode):
+    """The base class sampling from scipy.stats distributions.
+
+    Attributes
+    ----------
+    dist_class : scipy.stats distribution object
+        The distribution from which to sample.
+    _rng : numpy.random._generator.Generator
+        This object's random number generator.
+    sample_size : tuple
+        The shape of the array to generate for each sample. The actual returned value
+        will be (num_samples, *size). If an empty tuple will generate a single value per sample.
+    params_names : list of str
+        The names of the parameters used to create the distribution object.
+
+    Parameters
+    ----------
+    dist_name : str
+        The name of the distribution from which to sample.
+    seed : int, optional
+        The seed to use.
+    node_label : str, optional
+        An optional label for the node. If not provided, a default label
+        will be created based on the function name and parameters.
+    **kwargs : dict, optional
+        The parameters to use to create the distribution object.
+    """
+
+    def __init__(self, dist_name, seed=None, node_label=None, **kwargs):
+        super().__init__(self._non_func, node_label=node_label)
+
+        # Create the template of the distribution class and save the parameters.
+        if not hasattr(scipy.stats, dist_name):
+            raise ValueError(f"The distribution {dist_name} is not supported.")
```

Comment:
Maybe saying that is not found under scipy.stats?

Reply:
Done